key now found

I get the below post deployment error from

clowderapp.yaml
https://paste.ubuntu.com/p/dJPtPfN52p/

cloudenv.yaml
https://paste.ubuntu.com/p/x3bcYp4WNF/

bonfire.yaml
https://paste.ubuntu.com/p/jr3rzKTx8r/

    kubectl create namespace demo
    kubectl create namespace rhc
    kubectl apply -f saas-templates/clowdenv.yaml
    bonfire deploy -c bonfire/config.yaml  tower -n demo -e tower


    2021-06-08 16:17:24 [   ERROR] [          MainThread] hit unexpected error!
    Traceback (most recent call last):
    File "/home/ladas1/.local/lib/python3.9/site-packages/bonfire/bonfire.py", line 644, in _cmd_config_deploy
        _wait_on_namespace_resources(ns, timeout)
    File "/home/ladas1/.local/lib/python3.9/site-packages/bonfire/bonfire.py", line 141, in _wait_on_namespace_resources
        wait_for_all_resources(namespace, timeout)
    File "/home/ladas1/.local/lib/python3.9/site-packages/bonfire/openshift.py", line 557, in wait_for_all_resources
        wait_for(
    File "/home/ladas1/.local/lib/python3.9/site-packages/wait_for/__init__.py", line 174, in wait_for
        out = func(*func_args, **func_kwargs)
    File "/home/ladas1/.local/lib/python3.9/site-packages/bonfire/openshift.py", line 510, in _all_resources_ready
        clowd_env_name = find_clowd_env_for_ns(namespace)["metadata"]["name"]
    TypeError: 'NoneType' object is not subscriptable
    ERROR: deploy failed
    make: *** [Makefile.minkube:147: minikube-up] Error 1

id probably upgrade that line to 

    yaml = clowd_env_name = find_clowd_env_for_ns(namespace)
    if 'metadata' in in yaml:
        if 'name' in yaml['metadata']:

--- a/bonfire/bonfire.py
+++ b/bonfire/bonfire.py
@@ -615,8 +615,9 @@ def _cmd_config_deploy(
         match = find_clowd_env_for_ns(ns)
         if not match:
             _error(
-                f"could not find a ClowdEnvironment tied to ns '{ns}'.  Specify one with "
-                "'--clowd-env' or apply one with 'bonfire deploy-env'"
+                f"could not find a ClowdEnvironment tied to ns '{ns}'.  Specify which one "
+                "if you have already deployed one with '--clowd-env' or deploy one with "
+                "'bonfire deploy-env'"
             )
         clowd_env = match["metadata"]["name"]
         log.debug("inferred clowd_env: '%s'", clowd_env)

--- a/bonfire/openshift.py
+++ b/bonfire/openshift.py
@@ -504,22 +504,23 @@ def wait_for_ready_threaded(waiters, timeout=300):
 def _all_resources_ready(namespace, timeout):
     already_waited_on = set()
 
-    # wait on ClowdEnvironment
+    # wait on ClowdEnvironment, if there's one using this ns as its targetNamespace
     start = time.time()
 
-    clowd_env_name = find_clowd_env_for_ns(namespace)["metadata"]["name"]
-    waiter = ResourceOwnerWaiter(namespace, "clowdenvironment", clowd_env_name)
-    if not waiter.wait_for_ready(timeout):
-        return False
+    clowd_env = find_clowd_env_for_ns(namespace)
+    if clowd_env:
+        waiter = ResourceOwnerWaiter(namespace, "clowdenvironment", clowd_env["metadata"]["name"])
+        if not waiter.wait_for_ready(timeout):
+            return False
 
-    for key in waiter.observed_resources:
-        already_waited_on.add(key)
+        for key in waiter.observed_resources:
+            already_waited_on.add(key)
 
     end = time.time()
     elapsed = end - start
     timeout = int(timeout - elapsed)
 
-    # wait on all ClowdApps
+    # wait on all ClowdApps in this namespace
     start = time.time()
 
     waiters = []

